modules/core/src/main/java/org/apache/synapse/config/SynapseConfiguration.java

```diff
@@ -1473,14 +1478,18 @@ public synchronized void destroy(boolean preserverState) {
 
         // destroy the Message processors
         for (MessageProcessor mp : messageProcessors.values()) {
-            mp.destroy();
+            ((AbstractMessageProcessor) mp).destroy(preserverState, isShuttingDown);
         }
 
         for (API api : apiTable.values()) {
             api.destroy();
         }
     }
 
+    public synchronized void destroy(boolean preserverState) {
```

Comment:
typo preserver? Here the input parameter is preserveState, but at line 1452 you pass the value of isShuttingDown. There's also another method introduced with both parameters preserveState and isShuttingDown which means that the 2 values could be different. This seems confusing to me. WDYT?